Allow folder creation through django admin

zachmullen opened the pull request:
FYI @danlamanna

brianhelba commented on dkc/core/admin/folder.py:
```diff
@@ -9,3 +11,15 @@ class FolderAdmin(admin.ModelAdmin):
+            if obj.parent:
+                obj.tree = obj.parent.tree
+            else:
+                obj.tree = Tree.objects.create(quota=request.user.quota)
```
What if a tree is already selected?

zachmullen replied:
What do you mean already selected? Tree is always a computed property, it can't be manually selected even from the admin page.

brianhelba replied:
Oh, I see it's `editable=False`. I thought we had a little more discretion to set this up manually in the admin.

brianhelba commented on dkc/core/admin/folder.py:
```diff
@@ -9,3 +11,15 @@ class FolderAdmin(admin.ModelAdmin):
+            else:
+                obj.tree = Tree.objects.create(quota=request.user.quota)
+                obj.tree.grant_permission(
+                    PermissionGrant(user_or_group=request.user, permission=Permission.admin)
```
What if we want to grant this to another user or not grant it at all?

zachmullen replied:
I'm fine to omit this, this is just an attempt to not have a 500 when creating a folder through the admin page.

brianhelba replied:
What will 500? I would expect a 403 somewhere at worst.

Though can't an admin already do anything, even without explicit permission?

zachmullen replied:
On current master, trying to create a new folder via the admin page gives a 500 because it attempts to save a folder with no tree attached. @danlamanna ran into this bug earlier today. We could maybe try to preserve purity by making people manually create a Tree object and then attach a folder to it, but personally I'd rather they always get created together atomically.

brianhelba replied:
Sorry, I just mean is it valid to omit `obj.tree.grant_permission`?

zachmullen replied:
Yes, I believe that's valid.

brianhelba reviewed (commented):
I think this is fine as a temporary usability measure, but if we start relying on this, it's an indication that we need to start building some server-rendered staff pages (which can provide a lot more safety and convenience than the admin). We shouldn't abstract away the power of developers to make any change that the models technically permit.

For example here, it should ideally be possible for a developer to create a tree with no permission grants, or granted to a different user. If there's not adequate admin tooling to manage permission grants on Trees, that's something we can improve more directly.

zachmullen commented:
I removed the permission grant and changed the quota to use the creator's quota rather than the request user, since the admin page makes administrators select a creator.

zachmullen commented on dkc/core/admin/folder.py:
```diff
@@ -9,3 +10,12 @@ class FolderAdmin(admin.ModelAdmin):
+            if obj.parent:
+                obj.tree = obj.parent.tree
+            else:
+                obj.tree = Tree.objects.create(quota=obj.creator.quota)
```
@brianhelba it might make sense to grant admin permission to `obj.creator`, but I'll let you decide.

brianhelba replied:
I agree this makes more sense.
